core/runner/src/main/java/com/cognifide/aet/runner/processing/data/SuiteComparator.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * AET
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.cognifide.aet.runner.processing.data;
+
+import com.cognifide.aet.communication.api.metadata.Comparator;
+import com.cognifide.aet.communication.api.metadata.Operation;
+import com.cognifide.aet.communication.api.metadata.Step;
+import com.cognifide.aet.communication.api.metadata.Suite;
+import com.cognifide.aet.communication.api.metadata.Test;
+import com.cognifide.aet.communication.api.metadata.Url;
+import com.google.common.collect.Sets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class SuiteComparator {
+
+  /**
+   * Takes collection of suites and generates info messages if provided suites differ from each
+   * other. Compares each suite with all other suites. Compares tests (by count and names), and
+   * urls/steps (by count, types, names and parameters).
+   *
+   * @returns List of messages outlining which suites/test differ from each other.
+   */
+  static List<String> compare(List<Suite> suites) {
+    if (suites.size() <= 1) {
+      return Collections.emptyList();
+    }
+
+    List<List<Suite>> suitesUniquePairs = findUniquePairs(suites);
+    List<String> comparisonMessages = new ArrayList<>();
+
+    suitesUniquePairs.forEach(pair -> compareSuitesPair(pair, comparisonMessages));
+
+    return comparisonMessages;
+  }
+
+  private static List<List<Suite>> findUniquePairs(List<Suite> suites) {
```

Comment:
Instead of creating list of lists, wouldn't be better to introduce new object i.e. 'Pair'?